### ISSUE TITLE
How to show multiple playerViews simultaneously with one simpleExoPlayer instance?

### ISSUE THREAD

**Issue by psycureaws** [google/ExoPlayer#9567]
I am developing a video streaming app with android.

And I found some problem while developing using exoplayer2.

I want to show multiple playerViews simultaneously with one simpleExoPlayer instance

But I couldn't assign the same simpleExoPlayer instance to multiple playerViews.

How to show multiple playerViews simultaneously with one simpleExoPlayer instance?

Below is the code I wrote.

    playerView = findViewById(R.id.player_view);
    playerViewL = findViewById(R.id.player_viewL);
    playerViewR = findViewById(R.id.player_viewR);
    // exoPlayer = new ExoPlayer().
    simpleExoPlayer1 = new SimpleExoPlayer.Builder(this).build();
    simpleExoPlayer2 = new SimpleExoPlayer.Builder(this).build();
    simpleExoPlayer3 = new SimpleExoPlayer.Builder(this).build();
    playerView.setPlayer(simpleExoPlayer1);  // PlayerView cannot play video by simpleExoPlayer1, I want to play both (and all after)
    playerViewL.setPlayer(simpleExoPlayer2); // PlayerView can play video by simpleExoPlayer2
    playerViewR.setPlayer(simpleExoPlayer1); // PlayerView only can play video by simpleExoPlayer1
    simpleExoPlayer1.setRepeatMode(simpleExoPlayer1.REPEAT_MODE_ALL);
    simpleExoPlayer2.setRepeatMode(simpleExoPlayer2.REPEAT_MODE_ALL);
    simpleExoPlayer3.setRepeatMode(simpleExoPlayer3.REPEAT_MODE_ALL);

I hope you have a simple way.

thank you

**Comment by claincly** [comment]
It's not the intended use case for ExoPlayer, and I don't think it will work. I would suggest using two player instances.

But I think I might be missing something here, so please specify what is the reason that you want to share a player across many player views?

**Comment by psycureaws** [comment]
yes, currently simpleExoPlayer instance is need per each playerView.
but I want to switching single-full-screen and double-half-screen mode with synced by normal 2d video sources.
so I want download video from server for just one stream and represent to over two playerView or one playerView with half-partition-copied display.
below is the code all excuted **but not synced.**


      simpleExoPlayer1 = new SimpleExoPlayer.Builder(this).build();
      simpleExoPlayer2 = new SimpleExoPlayer.Builder(this).build();
      simpleExoPlayer3 = new SimpleExoPlayer.Builder(this).build();
      playerView.setPlayer(simpleExoPlayer1);  // PlayerView cannot play video by simpleExoPlayer1, I want to play both (and all after)
      playerViewL.setPlayer(simpleExoPlayer2); // PlayerView can play video by simpleExoPlayer2
      playerViewR.setPlayer(simpleExoPlayer3); // PlayerView only can play video by simpleExoPlayer1
      simpleExoPlayer1.setRepeatMode(simpleExoPlayer1.REPEAT_MODE_ALL);
      simpleExoPlayer2.setRepeatMode(simpleExoPlayer2.REPEAT_MODE_ALL);
      simpleExoPlayer3.setRepeatMode(simpleExoPlayer3.REPEAT_MODE_ALL);

I hope told you detailed.
thank you

**Comment by claincly** [comment]
I believe with the code you provide you could **theoretically** play the three videos at the same time. It's only theoretical because the ability to play multiple videos at the same time depends on your device spec (how many hardware decoders does it support).

Furthermore, it's not possible to sync multiple ExoPlayer instances at the moment, see #9283.
